editors/code/src/client.ts

```diff
@@ -41,6 +49,15 @@ export function createClient(serverPath: string, cwd: string): lc.LanguageClient
         diagnosticCollectionName: "rustc",
         traceOutputChannel,
         middleware: {
+            provideDocumentSemanticTokens(document: vscode.TextDocument, token: vscode.CancellationToken, next: DocumentSemanticsTokensSignature): vscode.ProviderResult<vscode.SemanticTokens> {
+                return semanticHighlightingWorkaround(next, document, token);
+            },
+            provideDocumentSemanticTokensEdits(document: vscode.TextDocument, previousResultId: string, token: vscode.CancellationToken, next: DocumentSemanticsTokensEditsSignature): vscode.ProviderResult<vscode.SemanticTokensEdits | vscode.SemanticTokens> {
+                return semanticHighlightingWorkaround(next, document, previousResultId, token);
+            },
+            provideDocumentRangeSemanticTokens(document: vscode.TextDocument, range: vscode.Range, token: vscode.CancellationToken, next: DocumentRangeSemanticTokensSignature): vscode.ProviderResult<vscode.SemanticTokens> {
```

Comment:
I'm not a typescript expert but do these need to be `async`? Might be worth a review from @Veetaha

Reply:
They only need to be `async` if we're using `await` somewhere in the function. In this case, since we're only passing on the result of the underlying function (which _is_ `async` and so returns a `Promise`), there's no need for these functions themselves to be `async`.